### ISSUE TITLE
Launching Dev Tools - Timeout error

### ISSUE THREAD

**Issue by kayne87** [facebook/react-native#19642]
- [x] Review the documentation: https://facebook.github.io/react-native
  - [x] Search for existing issues: https://github.com/facebook/react-native/issues
  - [x] Use the latest React Native version: https://github.com/facebook/react-native/releases
  - [x] Run `react-native info` in your terminal and paste its contents under "Environment"
  - [x] Let us know how to reproduce the issue. Include a code sample, share a project, or
    share an app that reproduces the issue using https://snack.expo.io/

## Environment
Environment:
  OS: Linux 4.15
  Node: 10.2.1
  Yarn: Not Found
  npm: 6.1.0
  Watchman: 4.9.4
  Xcode: N/A
  Android Studio: Not Found

Packages: (wanted => installed)
  react: 16.3.1 => 16.3.1
  react-native: ~0.55.2 => 0.55.4

## Description
I have a problem with the Activate Debug JS button inside the dev menu on the phone. 

In case is usefull to know, my scenario is: Windows and Virtualbox with Ubuntu inside. In Windows I started an Android emulator and in the VBox I have all my sources and servers. I connected the emulator with ubuntu adb by an ssh tunneling (https://stackoverflow.com/questions/1754162/remote-debugging-with-android-emulator) so when I start the application and send it to the device I press ctrl+m and start the debugging JS. After all of these actions in console I can see that I cannot execute Dev Tool.

I noticed that in previous issues about it Android developers use to do the following command: adb reverse tcp:8081 tcp:8081 but in my case System returns to me the following error: **error: more than one device and emulator**

## Steps to Reproduce
- sudo react-native run-android
- when the emulator runs the app press ctrl+m 
- On the dev menu, opened with prev step, press on Activate JS Debugging

## Expected Behavior
A Chrome tab should open with my logs and possibility to debug but nothing happend. One error is displayed in terminal

## Actual Behavior
localhost:8081 is unreacheble

**In terminal appears the following log**:

01:22:17: Launching Dev Tools...
01:22:17: Google Chrome exited with error: Error: Exited with code 1
01:22:17: at ChildProcess. (/home/marco/dev//node_modules/react-native/node_modules/opn/index.js:86:8)
01:22:17: at Object.onceWrapper (events.js:273:13)
01:22:17: at ChildProcess.emit (events.js:182:13)
01:22:17: at maybeClose (internal/child_process.js:961:16)
01:22:17: at Socket.stream.socket.on (internal/child_process.js:380:11)
01:22:17: at Socket.emit (events.js:182:13)
01:22:17: at Pipe._handle.close [as _onclose] (net.js:596:12)

**On the smartphone the following screen**:

![image](https://user-images.githubusercontent.com/5654001/41215182-633dc43a-6d4f-11e8-8cf2-614465a4583a.png)

**Comment by juanlet** [comment]
Even if you can set it up it will be a world of pain and really slow. You need to open the debugging ports of your virtual machine to be able to communicate with the device. Do yourself a favour and make a dual ubuntu/windows installation. If not, the first solution willwork since I tried to do the same but I had to install ubuntu without virtualbox, I have a top notch pc but the debugging time was superslow, plus virtualbox doesnt support emulatorsinside it which is a super nice feature to have

**Comment by kayne87** [comment]
Thanks for the answer, I remade the issue with the correct template but I read only now your message. Ok, I will try as you said with a main Linux distro without Virtual Machine. Actually I used Virtual Box because the performance is very good with my PC but I noticed some limits as you said, like VT-x.

**Comment by juanlet** [comment]
I was really convinced as well to develop inside a virtual machine but as soon as I encounter these limitations I had to switch to the real OS......Until virtualbox doesn't allow emulation inside a virtual machine I will not consider it again as an option. My pc specs are Ryzen 1800x, 16 gb RAM, nvidia gtx 1060 6gb, 512 ssd, 100mb internet broadband....So clearly pc resources has nothing to do with this slow connection between VM and device

**Comment by kayne87** [comment]
Hi juanlet, In this evening installed a new Linux OS.
I have the same problem. I installed google chrome by sudo apt-get install google-chrome-stable
In the attachment I leave the screenshot.
In the new installation I created a simple test app with **create-reactive-native-app** and I started with **sudo npm start**

**The emulator loads an Android KitKat API 19**

![screenshot from 2018-06-12 00-29-41](https://user-images.githubusercontent.com/5654001/41260709-e339f990-6dd8-11e8-8cf8-3b7418f417b1.png)

**Comment by kayne87** [comment]
For greater completeness I noticed that the port 8081 is not busy.

![screenshot from 2018-06-12 00-50-13](https://user-images.githubusercontent.com/5654001/41261160-bbac5f74-6dda-11e8-9db3-be8a23f54b8b.png)

**Comment by kayne87** [comment]
Ok, I found it. I'm sorry for the multiple answers. I read a lot of outdated documentations.

I searched the opn/index.js in npm repository and I discovered that this package is usefull to open the browser from javascript. So I debugged the file and I saw that the link lunched was on the port 19001 and this is the link in which the dev server is running. As I understood now we have to open the page and the google chrome dev tools before to start the Remote JS Debugging, a different behaviour in respect I read before.

Uff, finally. And was so trivial after all...

Thank you very much.

Marco
